### PR TITLE
Update Docker for build

### PR DESCRIPTION
**Pull request by homdx**
- Fix Docker for build v7
- Update version deps packages

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -17,39 +17,54 @@ RUN set -ex && \
...
-ARG BOOST_VERSION=1_66_0
-ARG BOOST_VERSION_DOT=1.66.0
-ARG BOOST_HASH=5721818253e6a0989583192f96782c4a98eb6204965316df9f5ad75819225ca9
+ARG BOOST_VERSION=1_67_0
+ARG BOOST_VERSION_DOT=1.67.0
+ARG BOOST_HASH=2684c972994ee57fc5632e03bf044746f6eb45d4920c343937a465fd67a5adba
 RUN set -ex \
     && curl -s -L -o  boost_${BOOST_VERSION}.tar.bz2 https://dl.bintray.com/boostorg/release/${BOOST_VERSION_DOT}/source/boost_${BOOST_VERSION}.tar.bz2 \
-    && echo "${BOOST_HASH}  boost_${BOOST_VERSION}.tar.bz2" | sha256sum -c \
+    && echo "${BOOST_HASH} boost_${BOOST_VERSION}.tar.bz2" | sha256sum -c \
```
Comment (moneromooo-monero) [inline Dockerfile]: Spurious whitespace changes (and these are actually needed apparently)

File: Dockerfile [hunk 2]
```diff
@@ -93,6 +109,8 @@ RUN set -ex \
     && make check \
     && make install
 
+
+
```
Comment (moneromooo-monero) [inline Dockerfile]: Spurious

File: README.md [hunk 3]
```diff
@@ -448,6 +448,9 @@ Then you can run make as usual.
         # Get binaries
         docker cp monero-android:/opt/android/monero/build/release/bin .
 
+
+
+
```
Comment (moneromooo-monero) [inline README.md]: Spurious

File: README.md [hunk 4]
```diff
@@ -498,6 +501,10 @@ Installing a snap is very quick. Snaps are secure. They are isolated with all of
         # or in background
         docker run -it -d -v /monero/chain:/root/.bitmonero -v /monero/wallet:/wallet -p 18080:18080 monero
 
+* need 3 Gb space
```
Comment (moneromooo-monero) [inline README.md]: The build needs 3 GB space.

File: Dockerfile [hunk 5]
```diff
@@ -1,7 +1,7 @@
 # Multistage docker build, requires docker 17.05
 
 # builder stage
-FROM ubuntu:16.04 as builder
+FROM debian:jessie as builder
```
Comment (moneromooo-monero) [inline Dockerfile]: Why ?

File: README.md (hunk 3 above)
Comment (homdx) [reply README.md]: Fixed in next commit

File: Dockerfile (hunk 5 above)
Comment (homdx) [reply Dockerfile]: Checked under Ubuntu. In next commit revert to 16.04

File: Dockerfile (hunk 1 above)
Comment (homdx) [reply Dockerfile]: Added space in next commit

File: Dockerfile (hunk 2 above)
Comment (homdx) [reply Dockerfile]: Removed to enter in next commit

File: README.md (hunk 4 above)
Comment (homdx) [reply README.md]: Fixed in next commit

File: Dockerfile [hunk 6]
```diff
@@ -35,21 +49,22 @@ RUN set -ex \
...
 
 # OpenSSL
-ARG OPENSSL_VERSION=1.0.2n
-ARG OPENSSL_HASH=370babb75f278c39e0c50e8c4e7493bc0f18db6867478341a832a982fd15a8fe
+ARG OPENSSL_VERSION=1.1.0h
+ARG OPENSSL_HASH=5835626cde9e99656585fc7aaa2302a73a7e1340bf8c14fd635a62c66802a517
 RUN set -ex \
     && curl -s -O https://www.openssl.org/source/openssl-${OPENSSL_VERSION}.tar.gz \
-    && echo "${OPENSSL_HASH}  openssl-${OPENSSL_VERSION}.tar.gz" | sha256sum -c \
+    && echo "${OPENSSL_HASH} openssl-${OPENSSL_VERSION}.tar.gz" | sha256sum -c \
```
Comment (moneromooo-monero) [inline Dockerfile]: Still here.

File: Dockerfile [hunk 7]
```diff
@@ -17,14 +17,28 @@ RUN set -ex && \
...
 
 WORKDIR /usr/local
 
+#Cmake
+ARG CMAKE_VERSION=3.11.4
+ARG CMAKE_VERSION_DOT=v3.11
+ARG CMAKE_HASH=8f864e9f78917de3e1483e256270daabc4a321741592c5b36af028e72bff87f5
+RUN set -ex \
+    && curl -s -O https://cmake.org/files/${CMAKE_VERSION_DOT}/cmake-${CMAKE_VERSION}.tar.gz \
+    && echo "${CMAKE_HASH} cmake-${CMAKE_VERSION}.tar.gz" | sha256sum -c \
```
Comment (moneromooo-monero) [inline Dockerfile]: That one will also need two spaces to work with some tools.

**Comment by moneromooo-monero** [comment]
There's just the remaining added lines without any other change, then please squash and it's good to go.

**Comment by moneromooo-monero** [comment]
They're still here, but whatever. Not super important at this point. Just rebase/squash and we're good.

**Comment by moneromooo-monero** [comment]
It would be nice if you could rebase/squash very soon, there'll be a 0.12.3.0 and this could go in if ready.

**Comment by homdx** [comment]
I w merge latest 3 commits in one

**Comment by moneromooo-monero** [comment]
There's a  "Merge remote-tracking branch 'upstream/master' " in there. Rebaseing (rather than merging) should fix it.

**Comment by homdx** [comment]
I  pull a remote branch master to my fork

**Comment by moneromooo-monero** [comment]
Even worse now, there are two.
If you can't get them out, an easier way is to create another branch, then cherry-pick the good commits. Then push -f to the PR branch.

**Comment by homdx** [comment]
Recreate new pull request
https://github.com/monero-project/monero/pull/4093

**Comment by homdx** [comment]
Updated this
https://github.com/monero-project/monero/pull/4094

**Comment by moneromooo-monero** [comment]
Thanks, I approved the other, this can can be closed now.

**Comment by moneromooo-monero** [comment]
+duplicate

**Pull request closed without merging**